src/NATS.Client.KeyValueStore/NatsKVStore.cs

```diff
@@ -166,13 +174,34 @@ public ValueTask PurgeAsync(string key, NatsKVDeleteOpts? opts = default, Cancel
 
     /// <inheritdoc />
     public async ValueTask<NatsKVEntry<T>> GetEntryAsync<T>(string key, ulong revision = default, INatsDeserialize<T>? serializer = default, CancellationToken cancellationToken = default)
+    {
+        var result = await TryGetEntryAsync(key, revision, serializer, cancellationToken);
+        result.EnsureSuccess();
+        return result.Value;
+    }
+
+    /// <inheritdoc />
+#if !NETSTANDARD
+    [AsyncMethodBuilder(typeof(PoolingAsyncValueTaskMethodBuilder<>))]
+#endif
+    public async ValueTask<NatsResult<NatsKVEntry<T>>> TryGetEntryAsync<T>(string key, ulong revision = default, INatsDeserialize<T>? serializer = default, CancellationToken cancellationToken = default)
     {
         ValidateKey(key);
         serializer ??= JetStreamContext.Connection.Opts.SerializerRegistry.GetDeserializer<T>();
 
-        var request = new StreamMsgGetRequest();
+#if NET8_0_OR_GREATER
+        var keySubject = string.Create(key.Length + Bucket.Length + 5, (Bucket, key), static (span, state) =>
+        {
+            "$KV.".CopyTo(span);
+            state.Bucket.CopyTo(span[4..]);
+            span[state.Bucket.Length + 4] = '.';
+            state.key.CopyTo(span[(state.Bucket.Length + 5)..]);
+        });
+#else
         var keySubject = $"$KV.{Bucket}.{key}";
```

Comment:
Another opportunity to use `_keySubjectRequestBeginning` to lower copies even if we keep string.Create pattern, (BTW name is trash feel free to pick a better one :))